addon/components/power-select-infinity.js

```diff
@@ -0,0 +1,53 @@
+import Component from '@ember/component';
+import { computed } from '@ember/object';
+import layout from '../templates/components/power-select-infinity';
+import { scheduleOnce } from '@ember/runloop';
+
+export default Component.extend({
+  tagName: '',
+  layout,
+  tabindex: -1,
+  triggerComponent: 'power-select-infinity/trigger',
+  optionsComponent: 'power-select-infinity/options',
+  beforeOptionsComponent: null,
+  searchEnabled: false,
+  loadingMessage: null,
+  noMatchesMessage: null,
+
+  // CPs
+  concatenatedTriggerClasses: computed('triggerClass', function() {
+    let classes = ['ember-power-select-typeahead-trigger'];
+    let passedClass = this.get('triggerClass');
```

Comment:
@bakerac4 im assuming alot of this copied from the power-select but thoughts on switching to `Ember.get()`?